devnet-sdk/testing/testlib/validators/forks.go

```diff
@@ -0,0 +1,137 @@
+package validators
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ethereum-optimism/optimism/devnet-sdk/system"
+	"github.com/ethereum-optimism/optimism/devnet-sdk/testing/systest"
+	"github.com/ethereum-optimism/optimism/op-node/rollup"
+)
+
+// getForkConfig is a helper function that retrieves the ForkConfig for a specific L2 chain.
+func getForkConfig(t systest.T, sys system.System, chainIdx uint64) (system.ChainConfig, *uint64, error) {
```

Comment:
note: not your fault, but we really need to get rid of those chainIdx shortcuts. It puts the onus on the test writer to have specific expectations for an arbitrary L2, instead of finding one that works and returning it.

Reply:
Agreed, it would be better to have a rollup validator/getter where all requirements are specified.

Reply:
Might make sense to have hierarchical validators. i.e. validator that we have 2 rollups in a superchain, and then pass the rollup getters to another validator that will ensure it has properties required.